src/OpenSpaceToolkit/Astrodynamics/Trajectory/State/CoordinatesSubset.cpp

```diff
@@ -57,26 +57,24 @@ Size CoordinatesSubset::getSize() const
 
 VectorXd CoordinatesSubset::add(
     [[maybe_unused]] const Instant& anInstant,
-    const VectorXd& aFullCoordinatesVector,
-    const VectorXd& anotherFullCoordinatesVector,
+    [[maybe_unused]] const VectorXd& aFullCoordinatesVector,
+    [[maybe_unused]] const VectorXd& anotherFullCoordinatesVector,
     [[maybe_unused]] const Shared<const Frame>& aFrameSPtr,
-    const Shared<const CoordinatesBroker>& aCoordinatesBrokerSPtr
+    [[maybe_unused]] const Shared<const CoordinatesBroker>& aCoordinatesBrokerSPtr
 ) const
 {
-    return aCoordinatesBrokerSPtr->extractCoordinate(aFullCoordinatesVector, *this) +
-           aCoordinatesBrokerSPtr->extractCoordinate(anotherFullCoordinatesVector, *this);
+    throw ostk::core::error::runtime::Wrong("Addition operation");
 }
 
 VectorXd CoordinatesSubset::subtract(
     [[maybe_unused]] const Instant& anInstant,
-    const VectorXd& aFullCoordinatesVector,
-    const VectorXd& anotherFullCoordinatesVector,
+    [[maybe_unused]] const VectorXd& aFullCoordinatesVector,
+    [[maybe_unused]] const VectorXd& anotherFullCoordinatesVector,
     [[maybe_unused]] const Shared<const Frame>& aFrameSPtr,
-    const Shared<const CoordinatesBroker>& aCoordinatesBrokerSPtr
+    [[maybe_unused]] const Shared<const CoordinatesBroker>& aCoordinatesBrokerSPtr
 ) const
 {
```

Comment:
I tried this and read up a bit on it.

I don't think the [[nodiscard]] would help, as its purpose is to signal that the returned value of the function is not to be discarded.

It looks like we can create a madro to avoid notating every single param, but it's still kind of ugly (uglier if you ask me, see GPT suggestion below):

```cpp
#if defined(__GNUC__) || defined(__clang__)
    #define UNUSED(x) UNUSED_ ## x __attribute__((unused))
#elif defined(_MSC_VER)
    #define UNUSED(x) __pragma(warning(suppress: 4100)) x
#else
    #define UNUSED(x) x
#endif

void myFunction(int usedParam, int UNUSED(unusedParam1), char UNUSED(unusedParam2)) {
    // Your code that uses 'usedParam' but not 'unusedParam1' or 'unusedParam2'
}
```

Reply:
Yeah definitely uglier, let's keep as is then, that was just a potential suggestion to remove the duplicate on `maybe_unused`

Reply:
Feel free to resolve